app/controllers/follow_requests_controller.rb

```diff
@@ -23,9 +24,13 @@ def edit
   def create
     @follow_request = FollowRequest.new(follow_request_params)
     @follow_request.sender = current_user
-
+  
     respond_to do |format|
-      if @follow_request.save
+      if @follow_request.sender == @follow_request.recipient
+        flash.now[:alert] = "Sender can't follow themselves"
+        format.html { render :new, status: :unprocessable_entity }
+        format.json { render json: { error: "Sender can't follow themselves" }, status: :unprocessable_entity }
+      elsif @follow_request.save
```

Comment:
Better to stay focused on one issue at a time, but ok for now.

Reply:
Can you explain more over "stay focused on one issue at a time" please?

Reply:
Of course. What I meant is that each pull request shall be as small as possible. This allows quicker review and testing, and reduces noise when looking at the diffs.

This pull request should be for implementing pundit only.

My comment was mostly for future reference, for now it's good.